Fully built out v6 demo

evwilkin opened the pull request:
Towards #202 

This PR adds the following items to the seed app for the V6 demo:
- Dashboard page
- Resources page
- Inventory page
- Dark/light theme toggle
- RTL switch
- Updated favicon

https://pf6-alpha2-demo.surge.sh/

evwilkin committed:
converting html demo to react

evwilkin committed:
pathways card

evwilkin committed:
pull from card demos

evwilkin committed:
built final 2 cards

evwilkin committed:
lint cleanup

evwilkin committed:
bump versions

nicolethoen committed:
feat: add inventory and resource pages, get tests running

nicolethoen committed:
attempt to fix build

nicolethoen committed:
more build fixes

nicolethoen committed:
add dark theme switcher

nicolethoen committed:
fix build

andrew-ronaldson commented:
Noticed these issues below:
The Events card on the Dashboard screen gets cut off.
![image](https://github.com/patternfly/patternfly-react-seed/assets/28658548/d4263ff6-2e5a-4485-9e6d-6860218d35db)

On the same screen this utility classes don't change with dark theme
![image](https://github.com/patternfly/patternfly-react-seed/assets/28658548/d0e52a87-c097-4709-8190-36d97742dd8b)

Lastly, there is an inline style on the spans. I think this should be using `var(--pf-t--global--text--color--subtle)`
<img width="1593" alt="Screenshot 2024-02-22 at 4 16 31 PM" src="https://github.com/patternfly/patternfly-react-seed/assets/28658548/dc6947e5-49fb-4259-90bb-2f29954e6b46">

snyk-bot committed:
fix: upgrade @patternfly/react-core from 5.1.2 to 5.2.0

Snyk has created this PR to upgrade @patternfly/react-core from 5.1.2 to 5.2.0.

See this package in npm:
https://www.npmjs.com/package/@patternfly/react-core

See this project in Snyk:
https://app.snyk.io/org/patternfly-bD6TiY6PxAoojbR6oZkeJN/project/015b745d-6900-4ccd-bbd4-e0cfeae5d2de?utm_source=github&utm_medium=referral&page=upgrade-pr

Commit:
chore(penta-demo): PR feedback - updated colors per core recs

Commit:
fix lint & update snapshot

Commit:
added RTL switch

evwilkin commented:
@nicolethoen sorry but I've had no luck pushing anything to surge, either to the existing link or spinning up a new surge URL - I'm getting an error for not having correct permissions to reuse the link in the description, and just getting the "page not found" surge page when it shows it's successfully deployed to a new URL.  

That's the only blocker for this PR currently, updating the preview with the fixes to the comments above (and ultimately making sure our consumers have a link they can access the updated demo through).  Do you have any thoughts?

Commit:
update favicon

evwilkin commented:
@andrew-ronaldson thanks for the feedback, connected with @mcoker (added as reviewer) for suggestions on fixes and fixed the overflowing grid item && update the incorrect text colors (note - now using bold tag rather than another color per suggestion).

mcoker commented on src/app/Dashboard/RecommendationsCard.tsx:
```diff
@@ -0,0 +1,82 @@
+              spaceItems={{ default: 'spaceItemsXs' }}
+              alignItems={{ default: 'alignItemsCenter' }}
+            >
+              <span className="pf-v5-u-font-size-2xl pf-v5-u-primary-color-100">{item.count}</span>
```
Hmmmm.... I'm not sure what to do about these numbers. To keep the same style, you could use an inline style with `--pf-t--global--text--color--brand--default` for the blue and `--pf-t--global--font--size--2xl`?

mcoker commented on src/app/Dashboard/CardStatus.tsx:
```diff
@@ -0,0 +1,261 @@
+import CheckCircleIcon from '@patternfly/react-icons/dist/js/icons/check-circle-icon';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon';
+import global_success_color_100 from '@patternfly/react-tokens/dist/js/global_success_color_100';
+import global_danger_color_100 from '@patternfly/react-tokens/dist/js/global_danger_color_100';
```
These will be the old v5 colors. Looks like these are just used to set icon colors, which can be updated to use `<Icon>` with the `status` prop for the color instead.

mcoker reviewed (approved):
I think it looks fine, but I left some comments. I think the summary is something like:

* Any existing utility classes will be based off of the old global vars. Spot checking, I think the "Text" utilities will be the most problematic. Flex, sizing, and accessibility utilities are probably fine the way they are and likely won't be any different in v6.
* Any `--pf-v5-global[...]` vars will be from the old system and should ideally be replaced with `--pf-t--global[...]`. Looks like most of the v5 global vars reference in these files are for spacers, which is great, because the spacers are the same between v5 and v6, they just have a different `--pf-t--global` prefix.
* Looks like we're importing colors from react-tokens, and those are referencing our old v5 colors. But looks like they're used to set icon colors, which is covered in the next bullet.
* Any time we need to set a status color on an icon, we can use `<Icon status="...">`
* Instead of `.pf-v5-u-font-color-200` or `text.color_200` (`text` is a reference to the text utility)....
  * The `<Text>` component will give you small/grey text using the `small` variant.
  * The `--pf-t--global--text--color--subtle` token will turn text grey/secondary in color (which you're already using some places, but not all) if you want the text to be grey but not small.

evwilkin committed:
chore(demo): Core PR feedback

evwilkin committed:
chore(demo): update snapshot

evwilkin committed:
updated snapshots for timestamp

evwilkin commented:
Thanks @mcoker, just FYI I went back and made those updates based on your suggestions 👍 
@nicolethoen @andrew-ronaldson this demo PR should be in a good place now after feedback & fixes.  The one outstanding obvious bug is the chart colors in dark theme, which I believe is yet to be updated for V6 so is as expected.

nicolethoen commented:
This just needs to be updated to the latest alphas as soon as the official alpha versions have been announced

nicolethoen committed:
Merge pull request #210 from patternfly/snyk-upgrade-ad708d72a30e61b2269ecdc187ae794f

[Snyk] Upgrade @patternfly/react-core from 5.1.2 to 5.2.0

wise-king-sullyman committed:
chore(deps): bump PF versions to latest alphas

wise-king-sullyman committed:
chore(deps): bump to latest PF alpha versions

wise-king-sullyman committed:
Merge branch 'main' into v6

safarcik committed:
fix: Patternfly brand name

dlabaj committed:
Merge pull request #221 from thybit/main

Fixed Patternfly brand name

dlabaj committed:
Merge branch 'main' into v6

dlabaj committed:
Merge pull request #192 from patternfly/v6

V6 penta demo

andrew-ronaldson commented:
<img width="1516" alt="Screenshot 2024-05-21 at 12 46 35 PM" src="https://github.com/patternfly/patternfly-react-seed/assets/28658548/547ae704-a089-4bc9-b11c-22e0a5a52503">

Thanks for the updates.
Seeing this blue bordered section on the page now.

nicolethoen committed:
fix: update to alpha2 versions

nicolethoen committed:
fix: update deps starting with PF

nicolethoen committed:
fix: update storybook deps

nicolethoen committed:
update lint & jest deps

nicolethoen committed:
more updates related to lint

nicolethoen committed:
update rest of minor version bumps

nicolethoen committed:
update lock file

nicolethoen committed:
Merge pull request #225 from nicolethoen/clean_up_main

Fix: Clean up dependencies in main branch

dependabot[bot] committed:
chore(deps): bump express from 4.18.0 to 4.19.2

Bumps [express](https://github.com/expressjs/express) from 4.18.0 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.0...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

nicolethoen committed:
Merge pull request #215 from patternfly/dependabot/npm_and_yarn/express-4.19.2

chore(deps): bump express from 4.18.0 to 4.19.2

nicolethoen committed:
fix: remove storybook

nicolethoen committed:
some fixes

nicolethoen committed:
Merge pull request #226 from nicolethoen/remove_storybook

fix: Remove storybook

nicolethoen committed:
feat: bump to lastest alpha2 - tests failing

nicolethoen committed:
bump to latest alphas

nicolethoen committed:
merge v6 branch into this demo

nicolethoen committed:
it's all working with alpha2

andrew-ronaldson commented:
That odd border still appears. Looks like it is from a `:visible-focus` on main 🤷‍♂️ . 
Since the labels in the demo are indicating a status we should use `p-m-danger/warning/success` etc. instead of `pf-m-red/green/orange` etc.
<img width="1343" alt="Screenshot 2024-06-27 at 9 12 58 AM" src="https://github.com/patternfly/patternfly-react-seed/assets/28658548/77946d4b-9a7c-4667-87be-ac7a71457850">

nicolethoen commented:
Latest surge preview with alpha2 versions: https://pf6-alpha2-demo.surge.sh/

evwilkin commented:
Closing as no longer needed.
